src/task_types/fuzzer_tasks.rs

```diff
@@ -0,0 +1,515 @@
+        self.sub_tasks.clone()
+    }
+}
+
+/// A service for generating fuzzer tasks.
+pub trait GenFuzzerService: Sync + Send {
+    /// Generate a fuzzer task.
+    fn generate_fuzzer_task(&self, params: &FuzzerGenTaskParams)
+        -> Result<Box<dyn GeneratedSuite>>;
+}
+
+/// Implementation of the GenFuzzerService.
+pub struct GenFuzzerServiceImpl {
+    /// Service to help generate multiversion test suites.
+    multiversion_service: Arc<dyn MultiversionService>,
+}
+
+impl GenFuzzerServiceImpl {
+    /// Create a new instance of the GenFuzzerService.
+    ///
+    /// # Arguments
+    ///
+    /// * `multiversion_service` - Service to help generate multiversion test suites.
+    pub fn new(multiversion_service: Arc<dyn MultiversionService>) -> Self {
+        Self {
+            multiversion_service: multiversion_service.clone(),
+        }
+    }
+}
+
+impl GenFuzzerService for GenFuzzerServiceImpl {
+    /// Generate a fuzzer task based on the given parameters.
+    ///
+    /// # Arguments
+    ///
+    /// * `params` - Parameters describing how to generate fuzzer.
+    ///
+    /// # Returns
+    ///
+    /// GeneratedSuite with details of how shrub task for the suite is built.
+    fn generate_fuzzer_task(
+        &self,
+        params: &FuzzerGenTaskParams,
+    ) -> Result<Box<dyn GeneratedSuite>> {
+        let task_name = &params.task_name;
+        let mut sub_tasks: Vec<EvgTask> = vec![];
+        if params.is_multiversion() {
+            let version_combinations = self
+                .multiversion_service
+                .get_version_combinations(&params.suite)?;
+            event!(
+                Level::INFO,
+                task_name = task_name.as_str(),
+                "Generating multiversion fuzzer"
+            );
+            for (old_version, version_combination) in self
+                .multiversion_service
+                .multiversion_iter(&version_combinations)
+            {
+                let base_task_name =
+                    build_name(&params.task_name, &old_version, &version_combination);
+                let base_suite_name = build_name(&params.suite, &old_version, &version_combination);
+
+                sub_tasks.extend(
+                    (0..params.num_tasks)
+                        .map(|i| {
+                            build_fuzzer_sub_task(
+                                &base_task_name,
+                                i,
+                                params,
+                                Some(&base_suite_name),
+                                Some(&version_combination),
+                            )
+                        })
+                        .collect::<Vec<EvgTask>>(),
+                );
+            }
+        } else {
+            sub_tasks = (0..params.num_tasks)
+                .map(|i| build_fuzzer_sub_task(&params.task_name, i, params, None, None))
+                .collect();
+        }
+
+        Ok(Box::new(FuzzerTask {
```

Comment:
A `Box` is a pointer to data. There are technical reasons why we are using it here. The summary is that we are going to treat it as a trait, but since traits are polymorphic, they need to be wrapped in some container.